timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/DbBoundManagedTimestampService.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.timelock.paxos;
+
+import com.palantir.timestamp.TimestampRange;
+import com.palantir.timestamp.TimestampService;
+
+public class DbBoundManagedTimestampService implements ManagedTimestampService {
+    private TimestampService timestampService;
+
+    public DbBoundManagedTimestampService(TimestampService timestampService) {
+
+        this.timestampService = timestampService;
+    }
+
+    @Override
+    public long getFreshTimestamp() {
+        return timestampService.getFreshTimestamp();
+    }
+
+    @Override
+    public TimestampRange getFreshTimestamps(int numTimestampsRequested) {
+        return timestampService.getFreshTimestamps(numTimestampsRequested);
+    }
+
+    @Override
+    public void fastForwardTimestamp(long currentTimestamp) {
+        throw new UnsupportedOperationException("Cannot fastforward timestamp on DB store");
```

Comment:
Yup, we dont expect the consumers of this to hit the `fastForward` endpoint.

Reply:
We still need this. Some of the consumers of this do actually do backup/restore the correct way instead of entirely via the db. Basically anyone using table splitting to have some tables in postgres and others in cassandra.

Reply:
Done.